flink-table/flink-table-planner/src/main/java/org/apache/flink/table/operations/AggregateOperationFactory.java

```diff
@@ -0,0 +1,468 @@
+
+		if (timeFieldExprs.size() != 1) {
+			throw new ValidationException("The time attribute should be a single field reference.");
+		}
+
+		Expression timeFieldExpr = timeFieldExprs.get(0);
+		if (!(timeFieldExpr instanceof FieldReferenceExpression)) {
+			throw new ValidationException("A group window expects a time attribute for grouping.");
+		}
+
+		FieldReferenceExpression timeField = (FieldReferenceExpression) timeFieldExpr;
+		validateTimeAttributeType(timeField);
+		return timeField;
+	}
+
+	private void validateTimeAttributeType(FieldReferenceExpression timeField) {
+		TypeInformation<?> timeFieldType = timeField.getResultType();
+		if (isStreaming) {
+			validateStreamTimeAttribute(timeFieldType);
+		} else {
+			validateBatchTimeAttribute(timeFieldType);
+		}
+	}
+
+	private void validateBatchTimeAttribute(TypeInformation<?> timeFieldType) {
+		if (!(timeFieldType instanceof SqlTimeTypeInfo || timeFieldType == LONG_TYPE_INFO)) {
+			throw new ValidationException("A group window expects a time attribute for grouping " +
+				"in a batch environment.");
+		}
+	}
+
+	private void validateStreamTimeAttribute(TypeInformation<?> timeFieldType) {
+		if (!(timeFieldType instanceof TimeIndicatorTypeInfo)) {
+			throw new ValidationException("A group window expects a time attribute for grouping " +
+				"in a stream environment.");
+		}
+	}
+
+	private ResolvedGroupWindow validateAndCreateTumbleWindow(
+			TumbleWithSizeOnTimeWithAlias window,
+			String windowName,
+			FieldReferenceExpression timeField) {
+		ValueLiteralExpression windowSize = getAsValueLiteral(window.getSize(),
+			"A group window expects a size value literal.");
+		validateWindowIntervalType(timeField, windowSize.getType());
+
+		return ResolvedGroupWindow.tumblingWindow(
+			windowName,
+			timeField,
+			windowSize);
+	}
+
+	private ResolvedGroupWindow validateAndCreateSlideWindow(
+			SlideWithSizeAndSlideOnTimeWithAlias window,
+			String windowName,
+			FieldReferenceExpression timeField) {
+		ValueLiteralExpression windowSize = getAsValueLiteral(window.getSize(),
+			"A group window expects a size value literal.");
+		ValueLiteralExpression windowSlide = getAsValueLiteral(window.getSlide(),
+			"A group window expects a slide value literal.");
+
+		TypeInformation<?> windowSizeType = windowSize.getType();
+		if (windowSizeType != windowSlide.getType()) {
+			throw new ValidationException("Sliding window expects same type of size and slide.");
+		}
+
+		validateWindowIntervalType(timeField, windowSizeType);
+
+		return ResolvedGroupWindow.slidingWindow(
+			windowName,
+			timeField,
+			windowSize,
+			windowSlide);
+	}
+
+	private ResolvedGroupWindow validateAndCreateSessionWindow(
+			SessionWithGapOnTimeWithAlias window,
+			String windowName,
+			FieldReferenceExpression timeField) {
+		ValueLiteralExpression windowGap = getAsValueLiteral(window.getGap(), "Window expects a gap value literal.");
+
+		if (windowGap.getType() == INTERVAL_ROWS) {
```

Comment:
are other types checked before?

Reply:
You're right :( I missed checking for `IntervalType`, sorry...